### PR TITLE
#985 -- Set the arrow to render by default on CTA buttons without option set

### PR DESCRIPTION
**Pull request by paboden**
Referencing this last issue found after merging last PR - https://github.com/undp/design-system/pull/1129#issuecomment-1437504308

@theadamparker - I set the CTA button to use the arrow by default if no icon option is set. 
The arrow default was altered when the download and external link icon options were added. This PR changes it back to show the arrow by default.

### CODE REVIEW

**Review by theadamparker — CHANGES_REQUESTED** [review]
@paboden , this approach introduces a similar regression that appears in 5 components using the button. [See list here](https://www.chromatic.com/build?appId=63d9556e407b5e99e6e9d920&number=66).

These 5 changes are using the button with **type** `secondary` and the **For_Primary** should be `no arrow`.

And looking at this now, the **For_Primary** control was originally intended only for buttons with type `primary`. It would make more sense to simply rename the For_Primary control to "Icon". 

To summarize, requesting the following changes:
- Rename `For_Primary` to `Icon`
- Update [the five components](https://www.chromatic.com/build?appId=63d9556e407b5e99e6e9d920&number=66) currently using the button to `secondary` button `no arrow`

**Review by theadamparker — APPROVED** [review]

**Review by Jura — APPROVED** [review]
